runtime/heiko/src/lib.rs

```diff
@@ -973,14 +975,32 @@ impl DataProviderExtended<CurrencyId, TimeStampedPrice> for AggregatedDataProvid
     }
 }
 
-pub struct Decimal;
-impl DecimalProvider for Decimal {
+pub struct Currencies;
+impl CurrencyProvider<CurrencyId, CurrencyType> for Currencies {
+    fn get_currency_type(asset_id: &CurrencyId) -> CurrencyType {
```

Comment:
We just need to distinguish whether the token is a native token or from pallet_asset.

Reply:
okay, got it.